lib/mkACI.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+import json
+import os
+from optparse import OptionParser
+from shutil import copyfile
+import subprocess
+import sys
+
+
+def parse_opts():
+    usage = "usage: %prog [options] file.aci manifest custom_env paths..."
+    parser = OptionParser(usage=usage)
+    parser.add_option("--thin", dest="thin")
+    parser.add_option("--dnsquirks", dest="dnsquirks")
+    parser.add_option("--static", dest="static")
+    (options, args) = parser.parse_args()
+    if len(args) < 3:
+        sys.stderr.write("not enough arguments")
+        parser.print_help()
+        sys.exit(1)
+    return options, args
+
+
+def paths_from_graphs(graphs):
+    paths = {}
+    for graph in graphs:
+        with open(graph) as f:
+            while True:
+                path = f.readline()
+                if not path:
+                    break
+                paths[path.rstrip()] = True
+                f.readline()
+                count = f.readline()
+                for i in range(int(count)):
+                    f.readline()
+    return paths.keys()
+
+
+def add_mounts(manifestPath, storePaths, mountArgPath):
+    with open(manifestPath) as f:
+        manifest = json.load(f)
+    with open(mountArgPath, "w+") as f:
+        mount_points = manifest["app"]["mountPoints"]
+        for path in storePaths:
+            name = path.replace(".", "").replace("/", "").lower()
+            mount_points.append({
+                'name': name,
+                'path': path,
+                'readOnly': True
+               })
+            args = "--volume=%s,kind=host,source=%s "
+            f.write(args % (name, path))
+    with open(manifestPath, "w") as f:
+        json.dump(manifest, f, indent=4)
+
+
+def tar(file_list, archive):
+    # prefix everything with rootfs except "manifest" and "rootfs" itself
+    t = ['flags=rSh',
+         's,^,rootfs/,',
+         's,^rootfs/manifest,manifest,',
+         's,^rootfs/rootfs,rootfs,']
+    cmd = ["tar",
+           "--create",
+           "--file", archive,
+           "--null",
+           "--gzip",
+           "-T", "-",
+           "--transform", ";".join(t)]
+    print("$ " + " ".join(cmd))
+    proc = subprocess.Popen(cmd, stdin=subprocess.PIPE)
+    proc.communicate(input=str.encode("\0".join(file_list)))
+
+
+def build_aci(aci,
+              manifest,
+              custom_env,
+              closures,
+              thin=True,
+              static=False,
+              dnsquirks=False):
+    os.mkdir(os.path.dirname(aci))
+    nixstore = []
+    envfiles = []
+    if static:
```

Comment:
I did the same thing as the old code here: https://github.com/steveeJ/nix2aci/pull/5/files#diff-83899cfd5c3f6fe16d157e4299c8c684L104